icd_windows_hkr.c

```diff
@@ -63,30 +63,11 @@ typedef enum
         mem = NULL;                 \
     } while (0)
 
-static const char OPENCL_REG_SUB_KEY[] = "OpenCLDriverName";
-
-#ifndef _WIN64
-static const char OPENCL_REG_SUB_KEY_WOW[] = "OpenCLDriverNameWow";
-#endif
-
-// Do not free the memory returned by this function.
-static const char* GetOpenCLRegKeyName(void)
-{
 #ifdef _WIN64
-    return OPENCL_REG_SUB_KEY;
+static const char OPENCL_REG_SUB_KEY[] = "OpenCLDriverName";
 #else
-    // The suffix/substring "WoW" is meaningful only when a 32-bit
-    // application is running on a 64-bit Windows OS. A 32-bit application
-    // running on a 32-bit OS uses non-WoW names.
-    BOOL is_wow64;
-    if (IsWow64Process(GetCurrentProcess(), &is_wow64) && is_wow64)
-    {
-        return OPENCL_REG_SUB_KEY_WOW;
-    }
-
-    return OPENCL_REG_SUB_KEY;
+static const char OPENCL_REG_SUB_KEY[] = "OpenCLDriverNameWow";
```

Comment:
The logic that got removed from here is needed for 32 bit processes running on 64 bit windows.

Reply:
Looks like something got added to the master after I did a pull request. The file needs to be appended. Only REG_SZ needs to be changed to REG_MULTI_SZ here.

Reply:
I think this should be in a common file to both dxgk and the other file.